src/opentok-screen-sharing.js

```diff
@@ -307,10 +309,10 @@
 
   };
 
-  var start = function () {
+  var start = function (props) {
     _log(_logEventData.actionStart, _logEventData.variationAttempt);
     extensionAvailable(_this.extensionID, _this.extensionPathFF)
-      .then(_initPublisher)
+      .then(_initPublisher(props))
```

Comment:
I think it's better to do 
`.then(() => _initPublisher(props))` since if `extensionAvailable` is rejected, then you will call _initPublisher(props) and in this case you shouldn't.

Reply:
The current behavior does not initialize a publisher if the screensharing extension fails